backend/controller/dal/notify.go

```diff
@@ -51,6 +53,88 @@ type event struct {
 	Deleted json.RawMessage `json:"deleted,omitempty"`
 }
 
+type deploymentState struct {
+	Key         model.DeploymentKey
+	schemaHash  []byte
+	minReplicas int
+}
+
+func deploymentStateFromDeployment(deployment Deployment) (deploymentState, error) {
+	hasher := sha256.New()
+	data := []byte(deployment.Schema.String())
+	if _, err := hasher.Write(data); err != nil {
+		return deploymentState{}, fmt.Errorf("failed to hash schema: %w", err)
+	}
+
+	return deploymentState{
+		schemaHash:  hasher.Sum(nil),
+		minReplicas: deployment.MinReplicas,
+	}, nil
+}
+
+func (d *DAL) pollDeployments(ctx context.Context) {
+	logger := log.FromContext(ctx)
+	retry := backoff.Backoff{}
+
+	previousDeployments := make(map[string]deploymentState)
+
+	for {
+		delay := time.Millisecond * 500
```

Comment:
Do we want this to be configurable maybe?

Reply:
I vote for getting it in as is and adding an option if there's a need